lib/urldata.h

```diff
@@ -1750,6 +1750,7 @@ struct Curl_easy {
   iconv_t inbound_cd;          /* for translating from the network encoding */
   iconv_t utf8_cd;             /* for translating to UTF8 */
 #endif /* CURL_DOES_CONVERSIONS && HAVE_ICONV */
+  bool in_callback;            /* true while executing a callback */
```

Comment:
I would prefer to see this added to the `struct UrlState`, which is a sub-struct within Curl_easy. Simply because we tend to store state "things" there.

Reply:
:+1: